tensorizer/serialization.py

```diff
@@ -581,12 +586,12 @@ def _verify_hashes(name: str,
                 sha.update(mv)
                 sha_digest = sha.digest()
                 if sha_digest != hash['hash']:
-                    raise RuntimeError(
+                    raise HashMismatchError(
                         f"Tensor '{name}' failed SHA256 verification. "
                         f"Expected {hash['hash']}, got {sha_digest}."
                     )
             else:
-                raise RuntimeError(
+                raise HashMismatchError(
```

Comment:
Not sure if this quite qualifies as a `HashMismatchError` like the other two. It's probably more accurately described with a `ValueError` like other parsing & `.tensors` version errors.

Reply:
Agreed, changed to `ValueError`.